rpcserver.go

```diff
@@ -7735,12 +7778,44 @@ func (r *rpcServer) AddInvoice(ctx context.Context,
 			err)
 	}
 
-	// Convert the asset amount into a fixed-point.
-	assetAmount := rfqmath.NewBigIntFixedPoint(req.AssetAmount, 0)
+	invUnits := req.AssetAmount
+
+	// If the invoice was created over a satoshi amount, we need to
+	// calculate the units.
+	if satsMode {
+		invUnits = rfqmath.MilliSatoshiToUnits(
+			amtMsat, *askAssetRate,
+		).ToUint64()
```

Comment:
We always need to use `ScaleTo(0).` before calling `ToUint64()`, otherwise we get incorrect results!